src/Poseidon/Generator/Types.hs

```diff
@@ -1,38 +1,38 @@
-module Poseidon.Generator.Types (
-    IndWithAdmixtureSet (..),
-    AdmixtureSet (..),
-    PopulationWithFraction (..),
-    IndWithPosition (..),
-    SpatialTemporalPosition (..),
-    GenoEntry (..)
-) where
-
-import           Data.List                  (intercalate)
+module Poseidon.Generator.Types where
+
+import           Data.List      (intercalate)
 import           Poseidon.Janno
-import           SequenceFormats.Eigenstrat (GenoEntry (..))
 
-data IndWithAdmixtureSet = IndWithAdmixtureSet {
-      _admixInd  :: String
-    , _admixUnit :: String
-    , _admixSet  :: AdmixtureSet
+data IndAdmixpops = IndAdmixpops {
+      _indName   :: String
+    , _groupName :: String
+    , _popSet    :: [PopAdmixpops]
+} deriving (Show)
+
+data PopAdmixpops = PopAdmixpops {
+      _popName :: String
+    , _popFrac :: Rational
+    , _popInds :: [(String, Int)]
+} deriving (Show)
+
+data InIndAdmixpops = InIndAdmixpops {
```

Comment:
I must admit that these type names are pretty obfuscating to me, like "InPopAdmixpops" ??? Fine

Reply:
True - the type names are horrible. I'll try to make them more readable.